src/modules/users/users.controller.ts

```diff
@@ -16,9 +17,9 @@ const handleError = (res, error) => {
 };
 
 const addUser = async (req, res) => {
-  const { name, age, email, tel, role } = req.body;
+  const User: User = req.body;
```

Comment:
We name only Classes with Capital letter. this isn't a Class make it with lower case

Reply:
btw, are you sure that in this case client can't send you any other data besides User? 
answer is no, so we can't be sure that inside this variable stored user.
you don't need to create here variable at all, use req.body as argument for service method